service/history/api/reapplyevents/api.go

```diff
@@ -177,6 +177,7 @@ func Invoke(
 			reappliedEvents, err := eventsReapplier.ReapplyEvents(
 				ctx,
 				mutableState,
+				context.UpdateRegistry(ctx, mutableState),
```

Comment:
Note to reviewers: most of the changes in this PR are passing the update registry down the stack to the event-reapplying code. I'll highlight the actual fix with another comment.